flytekit/deck/deck.py

```diff
@@ -174,3 +174,39 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependency", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        try:
+            from flytekitplugins.deck.renderer import TableRenderer
+        except ImportError:
+            warning_info = "Plugin 'flytekit-deck-standard' is not installed. To display python dependency, install the plugin in the image."
+            logger.warning(warning_info)
+            return warning_info
+
+        import subprocess
+
+        import pandas as pd
+
+        try:
+            installed_packages = subprocess.check_output(["pip", "freeze"]).decode().split("\n")
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Error occurred while fetching installed packages: {e}")
+            return ""
```

Comment:
Have you tried this? It's better to avoid creating a subprocess.
```python
from pip._internal.operations import freeze

x = freeze.freeze()
for dependency in x:
   print(dependency)
```

Reply:
Using `pip._internal` itself seems a dangerous. Since it is internal, their API may break. I would not want to have to pin `pip` versions in `flytekit`.

Also using `pip` adds a direct dependency on `pip` itself. This is usually installed, but we should add it into our `pyproject.toml`.